lib/github-glue.ts

```diff
@@ -100,21 +101,52 @@ export class GitHubGlue {
      */
     public async addPRCc(pullRequestURL: string, cc: string):
         Promise<void> {
-        const id = cc.match(/(<.*>)/);
+        const id = cc.match(/<(.*)>/);
 
         if (!id || id[1].match(/gitster@pobox\.com/)) {
             return;
         }
 
+        const ccLower = id[1].toLowerCase();
         const url = GitGitGadget.parsePullRequestURL(pullRequestURL);
         const pr = await this.getPRInfo(url[0], url[2]);
-        const ccNow = pr.body.match(/\r?\n\r?\n(cc:.*)/is);
+        const bodyEnd = pr.body.split(/\r?\n\s*?\r?\n/).pop();
+
+        let found = false;
+        let footerSeparator = "\r\n";
+        const noFooters = "No footers";
+
+        if (bodyEnd) try {
+            bodyEnd.trimRight().split(/\r?\n/).reverse().forEach(line => {
+                const match = line.match(/^([a-z][-a-z0-9]+):\s*(.*)$/i);
+
+                if (!match) {       // stop if not a footer
+                    throw new Error(noFooters);
+                }
+
+                footerSeparator = ""; // body already has footers
+                if (match[1].toLowerCase() === "cc") {
+                    addressparser(match[2], {flatten: true}).forEach(email => {
+                        if (ccLower === email.address.toLowerCase()) {
+                            found = true;
+                            throw new Error("Found");
```

Comment:
While it is quite unlikely for a last paragraph containing a line of the form `Cc: <exactly the address we're looking for>`, I would feel much better if we verified that we are really looking at footers, by verifying that every line begins with `<key>:`.